app/bundles/EmailBundle/Model/EmailModel.php

```diff
@@ -1770,18 +1833,34 @@ public function getEmailsLineChartData($unit, \DateTime $dateFrom, \DateTime $da
             if (!$canViewOthers) {
                 $this->limitQueryToCreator($q);
             }
+            if ($companyId !== null) {
+                $q->innerJoin('t', MAUTIC_TABLE_PREFIX.'companies_leads', 'company_lead', 't.lead_id = company_lead.lead_id')
+                    ->andWhere('company_lead.company_id = :companyId')
+                    ->setParameter('companyId', $companyId);
+            }
+            if ($campaignId !== null) {
+                $q->innerJoin('t', MAUTIC_TABLE_PREFIX.'campaign_events', 'ce', 't.source_id = ce.id AND t.source = "campaign.event"')
+                    ->innerJoin('ce', MAUTIC_TABLE_PREFIX.'campaigns', 'campaign', 'ce.campaign_id = campaign.id')
+                    ->andWhere('ce.campaign_id = :campaignId')
+                    ->setParameter('campaignId', $campaignId);
+            }
+            if ($segmentId !== null) {
+                $q->innerJoin('t', MAUTIC_TABLE_PREFIX.'lead_lists', 'll', 't.list_id = ll.id')
+                    ->andWhere('t.list_id = :segmentId')
+                    ->setParameter('segmentId', $segmentId);
+            }
             $data = $query->loadAndBuildTimeData($q);
 
             $chart->setDataset($this->translator->trans('mautic.email.clicked'), $data);
         }
 
         if ($flag == 'all' || $flag == 'unsubscribed') {
-            $data = $this->getDncLineChartDataset($query, $filter, DoNotContact::UNSUBSCRIBED, $canViewOthers);
+            $data = $this->getDncLineChartDataset($query, $filter, DoNotContact::UNSUBSCRIBED, $canViewOthers, $companyId, $campaignId, $segmentId);
             $chart->setDataset($this->translator->trans('mautic.email.unsubscribed'), $data);
         }
 
         if ($flag == 'all' || $flag == 'bounced') {
-            $data = $this->getDncLineChartDataset($query, $filter, DoNotContact::BOUNCED, $canViewOthers);
+            $data = $this->getDncLineChartDataset($query, $filter, DoNotContact::BOUNCED, $canViewOthers, $companyId, $campaignId, $segmentId);
```

Comment:
New params to existing methods must be nullable to respect BC. Fix on all places.

Reply:
They are

Reply:
My bad. What was I thinking...